src/NServiceBus.Persistence.AzureStorage/SagaPersisters/AzureSagaPersister.cs

```diff
@@ -101,6 +99,29 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                     }
                 }
             }
+
+            await RemoveSecondaryIndex(sagaData, table).ConfigureAwait(false);
```

Comment:
The secondary has a different partition key, based on the `propertyName + propertyValue`. That's the reason for the secondary :)